pkg/stanza/fileconsumer/internal/tracker/tracker.go

```diff
@@ -144,6 +155,26 @@ func (t *fileTracker) TotalReaders() int {
 	return total
 }
 
+func (t *fileTracker) restoreArchiveIndex() {
+	if !t.archiveEnabled() {
+		return
+	}
```

Comment:
nit: let's check this before calling the function. Maybe just my preference but it makes more sense to me to say "if it's enable, do it" rather than "do it. is it enabled?"